libminifi/include/processors/HashContent.h

```diff
@@ -0,0 +1,196 @@
+    HashReturnType ret_val;
+    ret_val.second = 0;
+    uint8_t buffer[HASH_BUFFER_SIZE];
+    MD5_CTX context;
+    MD5_Init(&context);
+
+    size_t ret = 0;
+    do {
+      ret = stream->readData(buffer, HASH_BUFFER_SIZE);
+      if(ret > 0) {
+        MD5_Update(&context, buffer, ret);
+        ret_val.second += ret;
+      }
+    } while(ret > 0);
+
+    if (ret_val.second > 0) {
+      unsigned char digest[MD5_DIGEST_LENGTH];
+      MD5_Final(digest, &context);
+      ret_val.first = digestToString(digest, MD5_DIGEST_LENGTH);
+    }
+    return ret_val;
+  }
+
+  HashReturnType SHA1Hash(const std::shared_ptr<org::apache::nifi::minifi::io::BaseStream>& stream) {
+    HashReturnType ret_val;
+    ret_val.second = 0;
+    uint8_t buffer[HASH_BUFFER_SIZE];
+    SHA_CTX context;
+    SHA1_Init(&context);
+
+    size_t ret = 0;
+    do {
+      ret = stream->readData(buffer, HASH_BUFFER_SIZE);
+      if(ret > 0) {
+        SHA1_Update(&context, buffer, ret);
+        ret_val.second += ret;
+      }
+    } while(ret > 0);
+
+    if (ret_val.second > 0) {
+      unsigned char digest[SHA_DIGEST_LENGTH];
+      SHA1_Final(digest, &context);
+      ret_val.first = digestToString(digest, SHA_DIGEST_LENGTH);
+    }
+    return ret_val;
+  }
+
+  HashReturnType SHA256Hash(const std::shared_ptr<org::apache::nifi::minifi::io::BaseStream>& stream) {
+    HashReturnType ret_val;
+    ret_val.second = 0;
+    uint8_t buffer[HASH_BUFFER_SIZE];
+    SHA256_CTX context;
+    SHA256_Init(&context);
+
+    size_t ret ;
+    do {
+      ret = stream->readData(buffer, HASH_BUFFER_SIZE);
+      if(ret > 0) {
+        SHA256_Update(&context, buffer, ret);
+        ret_val.second += ret;
+      }
+    } while(ret > 0);
+
+    if (ret_val.second > 0) {
+      unsigned char digest[SHA256_DIGEST_LENGTH];
+      SHA256_Final(digest, &context);
+      ret_val.first = digestToString(digest, SHA256_DIGEST_LENGTH);
+    }
+    return ret_val;
+  }
+}
+
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+static const std::map<std::string, const std::function<HashReturnType(const std::shared_ptr<io::BaseStream>&)>> HashAlgos =
+  { {"MD5",  MD5Hash}, {"SHA1", SHA1Hash}, {"SHA256", SHA256Hash} };
+
+//! HashContent Class
+class HashContent : public core::Processor {
+ public:
+  //! Constructor
+  /*!
+  * Create a new processor
+  */
+  explicit HashContent(std::string name,  utils::Identifier uuid = utils::Identifier())
+  : Processor(name, uuid)
+  {
+    logger_ = logging::LoggerFactory<HashContent>::getLogger();
+  }
+  //! Processor Name
+  static constexpr char const* ProcessorName = "HashContent";
+  //! Supported Properties
+  static core::Property HashAttribute;
+  static core::Property HashAlgorithm;
+  //! Supported Relationships
+  static core::Relationship Success;
+
+  void onSchedule(core::ProcessContext *context, core::ProcessSessionFactory *sessionFactory) override;
```

Comment:
I'm not sure how I missed this before, but we can't use override in libminifi due to some of the compilers in the environments we support not having that support yet in their version of GCC. It's used in extensions, but we kind of refrain from using this in libminifi until we can determine that those environments have acquiesced to later standards.

Reply:
Okay, removed them.